ingress/controllers/nginx/controller.go

```diff
@@ -41,14 +44,37 @@ import (
 )
 
 const (
-	defUpstreamName = "upstream-default-backend"
-	defServerName   = "_"
+	defUpstreamName          = "upstream-default-backend"
+	defServerName            = "_"
+	namedPortAnnotation      = "kubernetes.io/ingress-named-ports"
+	podStoreSyncedPollPeriod = 1 * time.Second
+	rootLocation             = "/"
 )
 
 var (
 	keyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type namedPortMapping map[string]string
+
+func (npm namedPortMapping) getPort(name string) (string, bool) {
+	val, ok := npm.getMappings()[name]
+	return val, ok
+}
+
+func (npm namedPortMapping) getMappings() map[string]string {
```

Comment:
please add a comment about what mappings, since the name is very general and it returns a map. Or rename the method.